### ISSUE TITLE
Identify and fix GraphQL subscriptions sometimes not working

### ISSUE THREAD

**Issue by Jannis** [graphprotocol/graph-node#332]
No entity changes seem to be logged.

1. Verify that there are any notifications in Postgres.
2. Verify whether the entity change listener thread is somehow broken.
3. Verify why change events are not being logged.
and so on.

**Comment by Jannis** [comment]
This is missing steps to reproduce.

**Comment by davekaj** [comment]
Got this error:

```
component: SubscriptionServer
 Sep 24 13:56:18.474 WARN Failed to establish WebSocket connection: WebSocket protocol error: Handshake not finished
 Sep 24 13:56:39.323 WARN Failed to establish WebSocket connection: WebSocket protocol error: Handshake not finished
component: RuntimeHostBuilder
 component: RuntimeHost
  Sep 24 13:56:48.826 ERRO Event subscription failed: RPC error: Transport error: Incomplete
  Sep 24 13:56:48.828 INFO shutting down WASM runtime, data_source: ENSRegistrar
```

This is the command I ran 

```
cargo run -p graph-node --release --   --postgres-url postgresql://davidkajpust@localhost:5432/ens-subgraph   --ethereum-rpc mainnet:http://127.0.0.1:8545   --ipfs 127.0.0.1:5001   --subgraph QmP45eBv5iJ1Lb77co1MWZntDuhqpGcHYRUPp8QFjuraGJ
```
Sometime seems to be wrong with the subscriptions. Note that when I turned the node off and then turned it back on, it worked for about 30 seconds again, and then failed with the same error message

**Comment by Jannis** [comment]
@davekaj There may be a confusion here: the issue is about GraphQL subscriptions. Even though the first two warnings are about GraphQL subscriptions, the error you posted is about Ethereum event subscriptions failing. The latter is a problem that'll be solved by the block stream work.

**Comment by davekaj** [comment]
Yes some confusion on my part here. I was talking to tim in slack and I ended up posting this here. Either way it sounds like the error I brought up will be fixed so all is good

**Comment by timmclean** [comment]
Oh whoops, that's my bad!

**Comment by Jannis** [comment]
@timmclean Nope. Again, this is about GraphQL subscriptions, which we still have to test more thoroughly.

**Comment by timmclean** [comment]
The `entity_changes_are_fired_and_forwarded_to_subscriptions` test in `store/postgres/tests/store.rs` is failing (and marked `#[ignore]` for now). Adding that to this issue since it's probably the same underlying problem.

**Comment by timmclean** [comment]
Reopening to check the test that was failing here:
https://github.com/graphprotocol/graph-node/blob/ce795ac67f0d7730bf87fb4c9a7ac3d25b05e4e8/store/postgres/tests/store.rs#L1351

**Comment by timmclean** [comment]
Some notes on my debugging so far:

The cause of the failing test wasn't particularly interesting (entity type name case didn't match). I've reproduced some subscriptions flakiness however on both `/subgraphs` and the decentraland subgraph.

To reproduce on `/subgraphs` (note: tested on `tim/subgraph-status` branch):
- dropdb && createdb
- Deploy decentraland subgraph to graph-node
- Open graphiql at `/subgraphs`
- Run a subscription query:
```
subscription {
  subgraphs {
    id
    entityCount
    createdAt
    status
    processedEthereumBlocksCount
    totalEthereumBlocksCount
    manifest {
      id,
      specVersion
      dataSources {
        id
        kind
        name
        network
      }
    }
  }
}
```
- The subscription will work, and the processedEthereumBlocksCount will update.
- Press the GraphiQL Execute Query button again
- Subscriptions will stop showing updates

On `/subgraphs/name/decentraland`, the following subscription query simply never returns updates:
```
subscription {
  parcels {
    id
    x
    y
    lastTransferredAt
  }
}
```
If I change `subscription` to `query` and rerun the query, I can see new results coming back.

I'll investigate further...

**Comment by timmclean** [comment]
Found it, I'll submit the 4-character fix tomorrow :joy: :sweat_smile:

**Comment by Jannis** [comment]
Yes! 😁

**Comment by timmclean** [comment]
I'll submit a PR in a minute to revive the entity changes test and fix the primary issue. I also just filed https://github.com/graphprotocol/graph-node/issues/647 for a secondary cause of subscriptions appearing flaky (it's bad UX, but lower priority than this issue in my opinion)
